test/cpp/jit/test_gpu.cpp

```diff
@@ -7857,6 +7940,10 @@ TEST_F(NVFuserTest, FusionReductionSchedulerNoODimShmoo_CUDA) {
 TEST_F(NVFuserTest, FusionReductionSchedulerDimShmoo_CUDA) {
   std::vector<DataType> dtypes = {
       DataType::Double, DataType::Float, DataType::Half};
+  // TODO: add complex support. Currently, complex fails with the following
+  // NVRTC compilation error:
+  //   error: no instance of overloaded function "__shfl_xor_sync" matches the
+  //   argument list
```

Comment:
We could disable lowering to warp shuffles for complex types. 
https://github.com/csarofeen/pytorch/blob/devel/torch/csrc/jit/codegen/cuda/lower_utils.cpp#L169-L215 is related code as well as:
https://github.com/csarofeen/pytorch/blob/devel/torch/csrc/jit/codegen/cuda/lower_warp_reduce.cpp#L401-L432

Reply:
Or, maybe transform `complex_tensor.sum(dim)` into `complex_tensor.view_as_real().sum(dim).view_as_complex()`, cross-referencing https://github.com/csarofeen/pytorch/pull/1481